homeworks/homework2/task2/memioze.py

```diff
@@ -0,0 +1,51 @@
+from functools import update_wrapper
+from typing import Callable, OrderedDict as OrderedDictType, Optional, Tuple, Dict, Any
+from collections import OrderedDict
+
+
+def memoize(function: Optional[Callable] = None, *, size: int = 0) -> Callable:
```

Comment:
I don't get why you create the function for initializing a class :) You can use `@FunctionMemoizer` as a decorator.